File: pset1/cash/cash.c
#include <math.h>
#include <cs50.h>
#include <stdio.h>

void process(void);

int main(void)
{
    process();

}
//this function makes alot, wooh, that was hard, thats hardvard
void process(void)
{
    int quarter = 25;
    int dime = 10;
    int nickel = 5;
    int pennie = 1;
    int cents;
    int remain;
    int divider;
    int total_divider;


    //prompt the user for input as dollars in loop to control negative number
    float dollars;

    do
    {
        dollars = get_float("How much change is owed in dollars:");
        cents = round(dollars * 100);
    }
    while (dollars <= 0);
    divider = 0;
    remain = 0;
    total_divider = 0;

    for (int i = 0; i < cents; i++)
    {



        if (cents >= quarter)
        {
            remain = cents % quarter;
            //calculate division
            divider = cents / quarter;
            total_divider = total_divider + divider;

        }
        else if (cents >= dime)
        {
            remain = cents % dime;
            //calculate division
            divider = cents / dime;
            total_divider = total_divider + divider;

        }
        else if (cents >= nickel)
        {
            remain = cents % nickel;
            //calculate division
            divider = cents / nickel;
            total_divider = total_divider + divider;

        }
        else
        {
            remain = cents % pennie;
            //calculate division
            divider = cents / pennie;
            total_divider = total_divider + divider;

        }

        cents = remain;


    }//for-loop curly bracket
    printf("%i\n", total_divider + cents);

}

File: pset6/cash/cash.py
from cs50 import get_float

quarter = 25
dime = 10
nickel = 5
pennie = 1
# //prompt the user for input as dollars in loop to control negative number
while (pennie == 1):
    dollars = get_float("Change owed:")
    if (dollars > 0):
        cents = int(dollars * 100)
        # print(cents)
        break


divider = 0
remain = 0
total_divider = 0
while (cents >= pennie):

    if (cents >= quarter):
        # print("this works")
        remain = cents % quarter
        # print(remain)
        # //calculate division
        divider = int(cents / quarter)
        # print(divider)
        total_divider = total_divider + divider
        # print(total_divider)

    elif (cents >= dime):
        remain = cents % dime
        # print(remain)
        # //calculate division
        divider = int(cents / dime)
        total_divider = total_divider + divider
        # print(total_divider)

    elif (cents >= nickel):
        remain = cents % nickel
        # //calculate division
        divider = int(cents / nickel)
        total_divider = total_divider + divider
        # print(total_divider)

    else:
        remain = cents % pennie
        # //calculate division
        divider = int(cents / pennie)
        total_divider = total_divider + divider
        # print(total_divider)

    cents = remain

print(f"{total_divider+cents}")

